src/App.js

```diff
@@ -4,33 +4,70 @@ import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 // component
 import NavBar from './components/NavBar';
 import Home from './components/Home';
-import LoginForm from './components/Login';
+import LoginForm from './components/User/Login';
 import AllTransactions from './components/AllTransactions'
 import AllAccounts from './components/AllAccounts';
 import UserOnboarding from './components/UserOnBoarding'
-import RegistrationForm from './components/Register';
+import RegistrationForm from './components/User/Register';
 import FriendsPage from './components/Friends';
+import SideBarComponent from './components/SideBar';
 
 // styles
 import { GlobalStyle } from './GlobalStyle';
+import AllFriendRequest from './components/FriendRequests';
+import AllScheduledTransactions from './components/AllScheduledTransactions';
+import AllSplitTransactions from './components/AllSplitTransactions';
 
-const App = () => (
-  <Router>
-    <NavBar />
-    <Routes>
-      <Route path='/' element={<LoginForm />} />
-      <Route path='/signup' element={<RegistrationForm />} />
-      <Route path='/onboarding' element={<UserOnboarding />} />
-      <Route path='/home' element={<Home />} />
-      <Route path='/expenses' element={<AllTransactions type='Expenses'/>} />
-      <Route path='/incomes' element={<AllTransactions type='Incomes'/>} />
-      <Route path='/accounts' element={<AllAccounts />} />
-      <Route path='/friends' element={<FriendsPage />} />
-
-    </Routes>
-    <GlobalStyle />
-  </ Router>
-)
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoggedIn: false
+    }
+  }
+
+  componentDidMount() {
+    this.setState({
+      isLoggedIn: localStorage.getItem('refresh') ? true : false
```

Comment:
the refresh token could exist and be expired, in that case you instead decode the token to check for expiry first